tests/contrib/openai/test_openai_v1.py

```diff
@@ -1346,3 +1346,100 @@ async def handle_async_request(self, request: httpx.Request) -> httpx.Response:
         assert False, f"Unexpected exception: {e}"
 
     assert asyncio_timeout, "Expected asyncio.TimeoutError"
+
+
+@pytest.mark.skipif(
+    parse_version(openai_module.version.VERSION) < (1, 66), reason="Response options only available openai >= 1.66"
+)
+def test_response(openai, openai_vcr):
+    """Ensure llmobs records are emitted for response endpoints when configured."""
+    with snapshot_context(
+        token="tests.contrib.openai.test_openai.test_response",
+        ignores=["meta.http.useragent", "meta.openai.api_type", "meta.openai.api_base"],
+    ):
+        with openai_vcr.use_cassette("response.yaml"):
+            model = "gpt-4.1"
+            input_messages = multi_message_input
+            client = openai.OpenAI()
+            client.responses.create(
+                model=model, input=input_messages, top_p=0.9, max_output_tokens=100, user="ddtrace-test"
+            )
+
+
+@pytest.mark.skipif(
+    parse_version(openai_module.version.VERSION) < (1, 66), reason="Response options only available openai >= 1.66"
+)
+def test_response_error(openai, openai_vcr):
+    """Assert errors when an invalid model is used."""
+    with snapshot_context(
+        token="tests.contrib.openai.test_openai.test_response_error",
+        ignores=["meta.http.useragent", "meta.openai.api_type", "meta.openai.api_base"],
+    ):
+        with openai_vcr.use_cassette("response_create_error.yaml"):
+            with pytest.raises(openai.BadRequestError):
+                client = openai.OpenAI()
+                client.responses.create(
+                    model="invalid-model",  # Using an invalid model to trigger error
+                    input="Hello world",
+                    user="ddtrace-test",
+                )
+
+
+@pytest.mark.skipif(
+    parse_version(openai_module.version.VERSION) < (1, 66), reason="Response options only available openai >= 1.66"
+)
+async def test_aresponse(openai, request_api_key, openai_vcr):
+    """Assert spans are created with async client."""
+    with snapshot_context(
```

Comment:
Can we use the `@pytest.mark.snapshot(token="...", ignores=[...])` decorator instead of inline context manager code?